tests/unit/abilities-api/wpRegisterAbility.php

```
@@ -465,4 +491,27 @@ public function test_get_all_registered_abilities() {
 		$result = wp_get_abilities();
 		$this->assertEquals( $expected, $result );
 	}
+
+	/**
+	 * Tests registering an ability with non-existent category.
+	 *
+	 * @expectedIncorrectUsage WP_Abilities_Registry::register
+	 */
+	public function test_register_ability_nonexistent_category(): void {
+		do_action( 'abilities_api_init' );
+
+		// Ensure category doesn't exist.
+		if ( WP_Abilities_Category_Registry::get_instance()->is_registered( 'nonexistent' ) ) {
```

Maybe assert here instead? It should never happen, so failing tests makes perfect sense.

---

Good catch! I've updated the test to use an assertion instead of the conditional check. 

The category 'nonexistent' should never be registered when this test runs, so if it does exist, it indicates a test isolation problem. Now the test will fail immediately with a clear message rather than silently working around the issue.

I also found and fixed two other similar cases:
- `test_ability_requires_existing_category` in wpAbilityCategory.php
- `test_filter_by_nonexistent_category` in wpRestAbilitiesListController.php

All three now use assertions to verify the test preconditions, which makes the tests more robust and the intent clearer.

Changes pushed in [6bde79a](https://github.com/WordPress/abilities-api/pull/102/commits/6bde79ad4648abb31c22b57a13f41d5afed62de3).